hs_app_timeseries/forms.py

```diff
@@ -143,38 +245,104 @@ class ProcessingLevelValidationForm(forms.Form):
+        kwargs.pop('cv_aggregation_statistics')
+        kwargs.pop('cv_statuses')
+
         super(TimeSeriesResultForm, self).__init__(*args, **kwargs)
-        self.helper = TimeSeriesResultFormHelper(allow_edit, res_short_id, element_id, element_name='TimeSeriesResult')
+        self.helper = TimeSeriesResultFormHelper(allow_edit, res_short_id, element_id,
+                                                 element_name='TimeSeriesResult')
+
+    def set_dropdown_widgets(self, current_sample_medium, current_units_type,
+                             current_agg_statistics, current_status):
+        cv_sample_medium_choices = _get_cv_dropdown_widget_items(self.cv_sample_mediums,
+                                                                 current_sample_medium)
+        self.fields['sample_medium'].widget = forms.Select(choices=cv_sample_medium_choices)
+        cv_units_type_choices = _get_cv_dropdown_widget_items(self.cv_units_types,
+                                                              current_units_type)
+        self.fields['units_type'].widget = forms.Select(choices=cv_units_type_choices)
+        cv_status_choices = _get_cv_dropdown_widget_items(self.cv_statuses, current_status)
+        self.fields['status'].widget = forms.Select(choices=cv_status_choices)
+        cv_agg_statistics_choices = _get_cv_dropdown_widget_items(self.cv_aggregation_statistics,
+                                                                  current_agg_statistics)
+        self.fields['aggregation_statistics'].widget = forms.Select(
+            choices=cv_agg_statistics_choices)
+
+    @property
+    def form_id(self):
+        form_id = 'id_timeseriesresult_%s' % self.number
+        return form_id
+
+    @property
+    def form_id_button(self):
+        form_id = 'id_timeseriesresult_%s' % self.number
+        return "'" + form_id + "'"
 
     class Meta:
         model = TimeSeriesResult
-        fields = ['units_type', 'units_name', 'units_abbreviation', 'status', 'sample_medium', 'value_count',
-                  'aggregation_statistics']
+        fields = ['units_type', 'units_name', 'units_abbreviation', 'status', 'sample_medium',
+                  'value_count', 'aggregation_statistics']
         widgets = {'value_count': forms.TextInput()}
 
+
 class TimeSeriesResultValidationForm(forms.Form):
     units_type = forms.CharField(max_length=255)
     units_name = forms.CharField(max_length=255)
     units_abbreviation = forms.CharField(max_length=20)
     status = forms.CharField(max_length=255)
     sample_medium = forms.CharField(max_length=255)
     value_count = forms.IntegerField()
-    aggregation_statistics = forms.CharField(max_length=255)
\ No newline at end of file
+    aggregation_statistics = forms.CharField(max_length=255)
+
+
+def _get_cv_dropdown_widget_items(dropdown_items, selected_item_name):
+    for item in dropdown_items:
+        if item.name == selected_item_name:
+            dropdown_items.remove(item)
```

Comment:
@pkdash should it ```break``` once it finds the item with selected_item_name? Is there a method that allows you to directly remove an item from dropdown_items by name to avoid this loop of checking each item in dropdown_items?